src/main/java/ru/skypro/homework/mapper/AdMapper.java

```diff
@@ -0,0 +1,60 @@
+package ru.skypro.homework.mapper;
+
+import org.springframework.stereotype.Component;
+import ru.skypro.homework.dto.AdDtoIn;
+import ru.skypro.homework.dto.AdDtoOut;
+import ru.skypro.homework.dto.AdExtendedDtoOut;
+import ru.skypro.homework.dto.AdsDtoOut;
+import ru.skypro.homework.entity.Ad;
+import ru.skypro.homework.entity.User;
+
+import java.util.List;
+
+@Component
+public class AdMapper {
+
+    //К Диме. Не вижу целесообразности в создании специального метода для преобразовании AdDtoIn в Ad
+    //Этот метод может потребоваться только один раз - при создании нового объявления
+    //и его текст может быть частью метода сервиса createAd.
+    //Обращаться к базе и вытягивать комментарии не стал - не знаю, для чего они.
+    public Ad toEntity(AdDtoIn adDtoIn) {
+        Ad ad = new Ad();
+        User user = new User(); //пока не рассказали, как добыть текущего пользователя
+        user.setId(1);
+        ad.setUser(user);
+        ad.setTitle(adDtoIn.getTitle());
+        ad.setPrice(adDtoIn.getPrice());
+        ad.setDescription(adDtoIn.getDescription());
+        return ad;
+    }
+    public AdDtoOut toAdDtoOut(Ad ad) {
+        AdDtoOut adDtoOut = new AdDtoOut();
+        adDtoOut.setPk(ad.getId());
+        adDtoOut.setAuthor(ad.getUser().getId());
+        adDtoOut.setTitle(ad.getTitle());
+        adDtoOut.setPrice(ad.getPrice());
+        //Вариант когда картинка лежит вместе с объявлением
+        adDtoOut.setImage("http://localhost:8080/ads/"+ad.getId()+"/image");
```

Comment:
```suggestion
        adDtoOut.setImage("/ads/"+ad.getId()+"/image");
```
http://localhost:8080 добавляется на фронте